### PR TITLE
[do not merge] mail-mta/sendmail: fixes for 8.16.1 plus cleanups

### PR DESCRIPTION
**Pull request by ionenwks**
As far as making it work again wrt [bug #730890](https://bugs.gentoo.org/730890), essential of it is just the first two commits.

Given that was trivial, felt could additionally do a bit of extra fixes/cleanups (not exhaustive but should make these easier to work with), should hopefully introduce no regressions.

Left out keyword restoration, be nice to get some feedback from the user from the bug and there's other things that worry me, e.g. the `$(<grep's ${EPREFIX}` that I left alone given I know little about what it's changing (all I did was convert leading spaces to tabs).

wrt [bug #681232](https://bugs.gentoo.org/681232) I think this covers everything that was left, so closing it.

### CODE REVIEW

File: mail-mta/sendmail/sendmail-8.16.1-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,203 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit systemd toolchain-funcs
```
Comment (thesamesam) [inline mail-mta/sendmail/sendmail-8.16.1-r1.ebuild:6]: Put a comment above inherit(?) for both sendmail and libmilter to mention bumping the other in sync where possible

File: mail-mta/sendmail/sendmail-8.16.1-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+	if use ldap; then
+		confMAPDEF+=" -DLDAPMAP"
+		confLIBS+=" -lldap -llber"
+	fi
+
+	if use sasl; then
+		confLIBS+=" -lsasl2"
+		confENVDEF+=" -DSASL=2"
+		confCCOPTS+=" -I/usr/include/sasl"
```
Comment (thesamesam) [inline mail-mta/sendmail/sendmail-8.16.1-r1.ebuild:65]: ${ESYSROOT} given it's passed directly with -I

File: mail-mta/sendmail/sendmail-8.16.1-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+		# add start-up options here
+		SENDMAIL_OPTS="-bd -q30m -L sm-mta" # default daemon mode
+		CLIENTMQUEUE_OPTS="-Ac -q30m -L sm-cm" # clientmqueue
+		KILL_OPTS="" # add -9/-15/your favorite evil SIG level here
+
+	EOF
+
+	if use sasl; then
+		dodir /etc/sasl2
+		cat <<- EOF > "${ED}"/etc/sasl2/Sendmail.conf || die "Sendmail.conf cat ailed"
```
Comment (thesamesam) [inline mail-mta/sendmail/sendmail-8.16.1-r1.ebuild:193]: 'Sendmail' vs sendmail looks odd, but if it's supposed to be that way, fair enough.

File: mail-mta/sendmail/sendmail-8.16.1-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+		confENVDEF+=" -DSASL=2"
+		confCCOPTS+=" -I/usr/include/sasl"
+		conf_sendmail_LIBS+=" -lsasl2"
+	fi
+
+	if use ssl; then
+		# Bug #542370 - lets add support for modern crypto (PFS)
+		confENVDEF+=" -DSTARTTLS -D_FFR_DEAL_WITH_ERROR_SSL"
+		confENVDEF+=" -D_FFR_TLS_1 -D_FFR_TLS_EC"
+		confLIBS+=" -lssl -lcrypto"
```
Comment (thesamesam) [inline mail-mta/sendmail/sendmail-8.16.1-r1.ebuild:73]: Ideally we'd png-config this.

File: mail-mta/sendmail/sendmail-8.16.1-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+	!mail-mta/opensmtpd
+	!mail-mta/postfix
+	!mail-mta/qmail-ldap
+	!>=mail-mta/ssmtp-2.64-r2[mta]
+	!net-mail/vacation"
+PDEPEND="!mbox? ( mail-filter/procmail )"
+
+src_prepare() {
+	eapply "${FILESDIR}"/${PN}-8.16.1-build-system.patch
+	eapply -p0 "${FILESDIR}"/sendmail-delivered_hdr.patch
```
Comment (thesamesam) [inline mail-mta/sendmail/sendmail-8.16.1-r1.ebuild:47]: ${PN} for consistency

File: mail-mta/sendmail/sendmail-8.16.1-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+
+inherit systemd toolchain-funcs
+
+DESCRIPTION="Widely-used Mail Transport Agent (MTA)"
+HOMEPAGE="https://www.sendmail.org/"
+SRC_URI="ftp://ftp.sendmail.org/pub/${PN}/${PN}.${PV}.tar.gz"
+
+LICENSE="Sendmail GPL-2" # GPL-2 is here for initscript
+SLOT="0"
+#KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86"
```
Comment (thesamesam) [inline mail-mta/sendmail/sendmail-8.16.1-r1.ebuild:14]: Add bug reference above this to explain why & how we'll restore keywords soon hopefully

**Review by thesamesam — COMMENTED** [review]
Let's ask the user on the bug to try this out?

File: mail-mta/sendmail/sendmail-8.16.1-r1.ebuild (hunk 2 above)
Comment (ionenwks) [reply mail-mta/sendmail/sendmail-8.16.1-r1.ebuild:65]: Good point, didn't want to touch too many things but these are obvious.

Comment (ionenwks) [reply mail-mta/sendmail/sendmail-8.16.1-r1.ebuild:65]: Then again, this is where this format gets nasty, e.g. this won't deal with spaces. Even if used an array it's latter seded into the .m4

Comment (ionenwks) [reply mail-mta/sendmail/sendmail-8.16.1-r1.ebuild:65]: Turns out this one is not needed, sendmail actually uses `include <sasl/...>` so I'll swap these for a tc-pkg --cflags libsasl2 along others I can.

File: mail-mta/sendmail/sendmail-8.16.1-r1.ebuild (hunk 3 above)
Comment (ionenwks) [reply mail-mta/sendmail/sendmail-8.16.1-r1.ebuild:199]: I agree, but I'd rather leave it how it was without looking deeper into it.

File: mail-mta/sendmail/sendmail-8.16.1-r1.ebuild [hunk 7]
```diff
@@ -35,98 +40,93 @@ RDEPEND="${DEPEND}
...
 src_compile() {
 	sh Build AR="$(tc-getAR)" RANLIB="$(tc-getRANLIB)" || die "compilation failed in main build script"
 }
 
 src_install() {
-	local MY_LIBDIR=/usr/$(get_libdir)
-	local MY_OBJDIR="obj.`uname -s`.`uname -r`.`uname -m`"
-
-	dodir /usr/bin ${MY_LIBDIR}
+	dodir /usr/{bin,$(get_libdir)}
```
Comment (Zlogene) [inline mail-mta/sendmail/sendmail-8.16.1-r1.ebuild:109]: Do we really need this? It is just a standard layout.

File: mail-mta/sendmail/sendmail-8.16.1-r1.ebuild [hunk 8]
```diff
@@ -35,98 +40,93 @@ RDEPEND="${DEPEND}
...
 	dodir /usr/share/man/man{1,5,8} /usr/sbin /usr/share/sendmail-cf
 	dodir /var/spool/{mqueue,clientmqueue} /etc/conf.d
```
Comment (Zlogene) [inline mail-mta/sendmail/sendmail-8.16.1-r1.ebuild:111]: and also why dodir /etc/conf.d?

File: mail-filter/libmilter/libmilter-1.0.2_p1-r2.ebuild [hunk 9]
```diff
@@ -18,60 +20,63 @@ S="${WORKDIR}/sendmail-${SENDMAIL_VER}"
...
-	local MY_LIBDIR=/usr/$(get_libdir)
-	dodir "${MY_LIBDIR}"
-	emake DESTDIR="${D}" LIBDIR="${MY_LIBDIR}" MANROOT=/usr/share/man/man \
-		SBINOWN=root SBINGRP=0 UBINOWN=root UBINGRP=0 \
-		LIBOWN=root LIBGRP=0 GBINOWN=root GBINGRP=0 \
-		MANOWN=root MANGRP=0 INCOWN=root INCGRP=0 \
-		MSPQOWN=root CFOWN=root CFGRP=0 \
-		MILTER_SOVER=${PV} \
-		install -C obj.*/libmilter
+	dodir /usr/$(get_libdir)
```
Comment (Zlogene) [inline mail-filter/libmilter/libmilter-1.0.2_p1-r2.ebuild:63]: ditto.

**Review by Zlogene — CHANGES_REQUESTED** [review]

**Comment by Zlogene** [comment]
oh wait, these dirs are a buildsystem fuck up, ignore me please.

**Pull request closed without merging**